pkg/sensors/exec/cgroups_test.go

```diff
@@ -317,3 +606,507 @@ func TestCgroupEventMkdirRmdir(t *testing.T) {
+	}
+}
+
+// Works in systemd Unified pure cgroupv2
+func TestCgroupv2K8sHierarchyInUnified(t *testing.T) {
+	testutils.CaptureLog(t, logger.GetLogger().(*logrus.Logger))
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	option.Config.HubbleLib = tus.Conf().TetragonLib
+	option.Config.Verbosity = 5
+
+	_, err := observer.GetDefaultObserver(t, ctx, tus.Conf().TetragonLib)
+	if err != nil {
+		t.Fatalf("GetDefaultObserver error: %s", err)
+	}
+
+	testManager := tus.StartTestSensorManager(ctx, t)
+	observer.SensorManager = testManager.Manager
+
+	testManager.EnableSensors(ctx, t, loadedSensors)
+	t.Cleanup(func() {
+		testManager.DisableSensors(ctx, loadedSensors)
+	})
+
+	// Probe full environment detection
+	setupTgRuntimeConf(t, invalidValue, invalidValue, invalidValue, invalidValue)
+	if cgroups.GetCgroupMode() != cgroups.CGROUP_UNIFIED {
+		logDefaultCgroupConfig(t)
+		t.Skipf("Skipping test %s as default cgroup mode is not Unified mode (Cgroupv2)", t.Name())
+	}
+
+	testDir := t.TempDir()
+	cgroupRoot, err := mountCgroup(t, testDir, "cgroup2", "")
+	if err != nil {
+		t.Fatalf("mountCgroup() failed: %v", err)
+	}
+
+	t.Cleanup(func() {
+		umountCgroup(t, cgroupRoot)
+	})
+
+	trackingCgrpLevel := uint32(0)
+	for i, c := range defaultKubeCgroupHierarchy {
+		if c.tracking == false {
+			trackingCgrpLevel = uint32(i)
+			break
+		}
+	}
+
+	require.NotZero(t, trackingCgrpLevel)
+	require.True(t, trackingCgrpLevel <= uint32(len(defaultKubeCgroupHierarchy)))
+
+	// Setup unified cgroup tracking 0 as hierarchy ID
+	setupTgRuntimeConf(t, trackingCgrpLevel, uint32(logrus.TraceLevel), 0, invalidValue)
+
+	logDefaultCgroupConfig(t)
+	logTetragonConfig(t, bpf.MapPrefixPath())
+	err = logCgroupMountInfo(t)
+	assert.NoError(t, err)
+
+	triggerCgroupMkdir := func() {
+		last := cgroupRoot
+		for _, dir := range defaultKubeCgroupHierarchy {
+			err = cgroupMkdir(t, last, "", dir.path)
+			if err != nil {
+				t.Fatalf("Failed to create cgroup %s/%s", last, dir.path)
+			}
+			last = filepath.Join(last, dir.path)
+		}
+	}
+
+	triggerCgroupRmdir := func() {
+		path := "/"
+		for _, dir := range defaultKubeCgroupHierarchy {
+			path = filepath.Join(path, dir.path)
+		}
+
+		for i := 0; i < len(defaultKubeCgroupHierarchy); i++ {
+			err := cgroupRmdir(t, cgroupRoot, "", path)
+			if err != nil {
+				t.Fatalf("Failed to remove cgroup %s/%s", cgroupRoot, path)
+			}
+			path = filepath.Dir(path)
+		}
+	}
+
+	triggers := []func(){
+		triggerCgroupMkdir, triggerCgroupRmdir,
+	}
+
+	t.Cleanup(func() {
+		cgroupRmdir(t, cgroupRoot, "", defaultKubeCgroupHierarchy[0].path)
+	})
+
+	testCgroupv2HierarchyInUnified(ctx, t, cgroupRoot, defaultKubeCgroupHierarchy,
+		trackingCgrpLevel, triggers)
+
+	assertCgroupDirTracking(t, defaultKubeCgroupHierarchy)
+}
+
+// Test Cgroupv2 tries to emulate k8s hierarchy without exec context
+// Works in systemd hybrid mode
+func TestCgroupv2K8sHierarchyInHybrid(t *testing.T) {
```

Comment:
This function seems very to `TestCgroupv2K8sHierarchyInUnified`. Can we just merge them so that we do DRY? 

```
@@ -1,4 +1,4 @@
-func TestCgroupv2K8sHierarchyInUnified(t *testing.T) {
+func TestCgroupv2K8sHierarchyInHybrid(t *testing.T) {
 	testutils.CaptureLog(t, logger.GetLogger().(*logrus.Logger))
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
 	defer cancel()
@@ -21,9 +21,9 @@
 
 	// Probe full environment detection
 	setupTgRuntimeConf(t, invalidValue, invalidValue, invalidValue, invalidValue)
-	if cgroups.GetCgroupMode() != cgroups.CGROUP_UNIFIED {
+	if cgroups.GetCgroupMode() != cgroups.CGROUP_HYBRID {
 		logDefaultCgroupConfig(t)
-		t.Skipf("Skipping test %s as default cgroup mode is not Unified mode (Cgroupv2)", t.Name())
+		t.Skipf("Skipping test %s as default cgroup mode is not Hybrid mode (Cgroupv1 and Cgroupv2)", t.Name())
 	}
 
 	testDir := t.TempDir()
@@ -89,7 +89,8 @@
 		cgroupRmdir(t, cgroupRoot, "", defaultKubeCgroupHierarchy[0].path)
 	})
 
-	testCgroupv2HierarchyInUnified(ctx, t, cgroupRoot, defaultKubeCgroupHierarchy,
+	// This test will run in hybrid mode since systemd will mount first cgroup2
+	testCgroupv2HierarchyInHybrid(ctx, t, cgroupRoot, defaultKubeCgroupHierarchy,
 		trackingCgrpLevel, triggers)
 
 	assertCgroupDirTracking(t, defaultKubeCgroupHierarchy)
```

Reply:
Done, thanks!